graphql_compiler/compiler/match_query.py

```diff
@@ -135,6 +136,34 @@ def _split_match_steps_into_match_traversals(match_steps):
     return output
 
 
+def _extract_global_operations(ir_blocks):
+    """Extract all global operation blocks (all blocks following GlobalOperationsStart).
```

Comment:
So is `ConstructResult` considered part of the global operations and returned in `global_operation_blocks`, or is it returned in `remaining_ir_blocks`?

From the use of this function below, it seems like `ConstructResult` shouldn't be part of `ir_blocks` at all -- in which case the docstring must reflect that, and there should be an assertion that ensures that.